homeassistant/components/wallbox/coordinator.py

```diff
@@ -150,6 +150,21 @@ async def async_set_charging_current(self, charging_current: float) -> None:
         )
         await self.async_request_refresh()
 
+    def _set_energy_cost(self, energy_cost: float) -> None:
+        """Set energy cost for Wallbox."""
+        try:
+            self._authenticate()
+            self._wallbox.setEnergyCost(self._station, energy_cost)
+        except requests.exceptions.HTTPError as wallbox_connection_error:
```

Comment:
In a follow-up PR please refactor all the protected functions as the all need authentication and have the same error handling.

Probably you could solve it with a decorator:
Could be something like this (untested)
```python
_WallboxCoordinatorT = TypeVar("_WallboxCoordinatorT", bound="WallboxCoordinator")
_P = ParamSpec("_P")

def _require_authentication(
    func: Callable[Concatenate[_WallboxCoordinatorT, _P], Any]
) -> Callable[Concatenate[_WallboxCoordinatorT, _P], Any]:
    """Decorator to authenticate first."""

    def require_authentication(
        self: _WallboxCoordinatorT, *args: _P.args, **kwargs: _P.kwargs
    ) -> Any:
        """Authenticate using Wallbox API."""
        try:
            self._wallbox.authenticate()
            return func(self, *args, **kwargs)
        except requests.exceptions.HTTPError as wallbox_connection_error:
            if wallbox_connection_error.response.status_code == HTTPStatus.FORBIDDEN:
                raise ConfigEntryAuthFailed from wallbox_connection_error
            raise ConnectionError from wallbox_connection_error

    return require_authentication

```